packages/scripts/src/flux-emulator/lib.ts

```diff
@@ -142,46 +159,111 @@ export const stop = async (inputs: Inputs): Promise<void> => {
   await setFluxConfig(newConfig, inputs.configServerSet).toPromise()
 }
 
+type IntegrationTestReducer = {
+  latestInput?: null | { data: Record<string, unknown> }
+  integrationTests: Record<string, unknown>[]
+}
+
 /**
  * Writes a json file for k6 to use as a payload based. Pulls the config from
  * weiwatchers to determine which adapter can hit which services and with which
  * pairs.
  * @param {Inputs} inputs The inputs to use to determine which adapter to create the config for
  */
 export const writeK6Payload = async (inputs: Inputs): Promise<void> => {
-  log(blue.bold('Fetching master config'))
-  const masterConfig = await fetchConfigFromUrl(inputs.weiWatcherServer).toPromise()
-  if (!masterConfig || !masterConfig.configs) {
-    process.exitCode = 1
-    throw red.bold('Could not get the master configuration')
-  }
+  logInfo('Fetching master config')
+  const masterConfig = await lastValueFrom(fetchConfigFromUrl(inputs.weiWatcherServer))
+  if (!masterConfig || !masterConfig.configs) throwError('Could not get the master configuration')
+
+  logInfo('Adding new adapter to qa config')
+  const newConfig: ReferenceContractConfig[] = addAdapterToConfig(
+    inputs.adapter,
+    inputs.ephemeralName,
+    masterConfig.configs as ReferenceContractConfig[],
+    [],
+  )
+
+  const configPayloads: ConfigPayload[] = newConfig.map(({ name, data }) => ({ name, data }))
+
+  // If no payloads from config, check integration tests
+  if (!configPayloads.length) {
+    let pathToAdapter = ''
+    const adapterTypes = ['sources', 'composites', 'targets']
+    for (const type of adapterTypes) {
+      const path = `packages/${type}/${inputs.adapter}`
+      if (shell.test('-d', path)) {
```

Comment:
Look into https://www.geeksforgeeks.org/node-js-fs-readdirsync-method/?

Reply:
@KuphJr I think I'll pass on `fs` for now since it wasn't clear to me how to test the type of a pathway. I saw something about `dirent` but you have to instantiate the class first.

Reply:
Sounds good!